frontend/ui/widget/touchmenu.lua

```diff
@@ -744,6 +746,105 @@ function TouchMenu:updateItems()
     end)
 end
 
+local MAX_MENU_DEPTH = 20 -- currently our menu needs at least 12 here
+function TouchMenu:_recurse(val, path, text, search_for, depth)
+    depth = depth + 1
+    if depth > MAX_MENU_DEPTH then
+        return
+    end
+    for i,v in pairs(val) do
+        local menu_text = nil
+
+        if type(v) == "table" and (type(i) == "number" or i:find("item")) then
+            local menu_entry = val[i].text or (val[i].text_func and  val[i].text_func())
+            local next_text = menu_entry and (text .. "→ " .. tostring(menu_entry)) or text -- drop sub_item_table
+            TouchMenu:_recurse(val[i], path .. "." .. i, next_text, search_for, depth)
+        elseif i == "text_func" then
+            menu_text = v()
+        elseif i == "text" then
+            menu_text = v
+        end
+
+        if menu_text and Utf8Proc.lowercase(menu_text):find(search_for) then
+            table.insert(TouchMenu.foundMenuItem, {menu_text, path .. "." .. i, text})
+        end
+    end
+end
+
+function TouchMenu:onMenuSearch(search_for)
+    TouchMenu.foundMenuItem = {}
```

Comment:
You would need to clean that results array after we are done with it, and not waiting until it is used again.

I see it is indeed cleaned at the end of `TouchMenu:onOpenMenu` - but may be not if we don't open any menu and just exit KVPage ?